chef/automate
# Config mgmt Get Nodes date filter

### :nut_and_bolt: Description: What code changed, and why?
Adding a start and end date to the GetNodes function to filter for when a node has last checked-in. 

### :chains: Related Resources

#3217 

### :athletic_shoe: How to Build and Test the Change
1. `build components/config-mgmt-service && build components/automate-gateway && start_all_services`
1. Add 3 nodes from now to 3 days ago 
```
for _ in {1..3} ; do
  date=$(date +%FT%TZ -d "$((RANDOM % 3)) day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw
done
```
1. Add 2 nodes from over 6 days ago. 
```
  date=$(date +%FT%TZ -d "6 day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw

  date=$(date +%FT%TZ -d "7 day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw
```
1. Filter for nodes from 5 days to now and ensure only 3 nodes are returned 
```
date=$(date +%FT%TZ -d "5 day ago")
curl -s -f --insecure -H "api-token: $(get_api_token)" "https://localhost/api/v0/cfgmgmt/nodes?pagination.page=1&pagination.size=100&start=$date" | jq .[].id | wc -l
```
1. Filter for nodes up to 4 days ago and ensure only 2 nodes are returned. 
```
curl -s -f --insecure -H "api-token: $(get_api_token)" "https://localhost/api/v0/cfgmgmt/nodes?pagination.page=1&pagination.size=100&end=$date" | jq .[].id | wc -l
```

### :white_check_mark: Checklist
